src/server/game/Spells/Spell.cpp

```diff
@@ -4973,6 +4973,23 @@ SpellCastResult Spell::CheckCast(bool strict)
         if (castResult != SPELL_CAST_OK)
             return castResult;
 
+        // If it's not a melee spell, check if vision is obscured by SPELL_AURA_INTERFERE_TARGETTING
+        if (m_spellInfo->DmgClass != SPELL_DAMAGE_CLASS_MELEE)
```

Comment:
I added this because it makes sense for smoke bomb to also be unaffected by melee attacks.
Unless you can provide proof of your claim, I can't make any more changes.